sylvia-derive/src/contract/communication/instantiate_builder.rs

```diff
@@ -0,0 +1,68 @@
+use convert_case::Case;
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::{GenericParam, Type, WhereClause};
+
+use crate::crate_module;
+use crate::parser::MsgType;
+use crate::types::msg_variant::MsgVariant;
+use crate::utils::{get_ident_from_type, SvCasing};
+
+pub struct InstantiateBuilder<'a> {
```

Comment:
It might. This is just how all the code looks like in the `sylvia-derive`.
The benefit is that in some cases we want to do some validation and data preparation in the `new` method which makes the `emit` method more readable.